pkg/controller/syncmachineset/syncmachineset_controller_test.go

```diff
@@ -0,0 +1,311 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syncmachineset
+
+import (
+	"fmt"
+	"testing"
+
+	cov1 "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+	coclient "github.com/openshift/cluster-operator/pkg/client/clientset_generated/clientset/fake"
+	informers "github.com/openshift/cluster-operator/pkg/client/informers_generated/externalversions"
+	"github.com/openshift/cluster-operator/pkg/controller"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	clientgofake "k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+
+	clusterapiv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clusterapiclient "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	clusterapiclientfake "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/fake"
+	clusterapiinformers "sigs.k8s.io/cluster-api/pkg/client/informers_generated/externalversions"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	testRegion           = "us-east-1"
+	testClusterVerUID    = types.UID("test-cluster-version")
+	testClusterVerName   = "v3-9"
+	testClusterVerNS     = "cluster-operator"
+	testClusterUUID      = types.UID("test-cluster-uuid")
+	testClusterName      = "testcluster"
+	testClusterNamespace = "testcluster"
```

Comment:
Should use a different name for namespace and name so that we can ensure that we aren't accidentally using one for the other in the code.